### DISCUSSION TITLE
详解 Kubernetes Deployment 的实现原理 · /kubernetes-deployment

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#550] [Announcements]
https://draveness.me/kubernetes-deployment

**Comment by draveness** [comment]
> 请问你用的是什么画图软件？

@wangcanfengxs 文章下面有关于图片的段落

**Comment by Devying** [comment]
作者你好，有没有用过weavescope 它的监控图里面的InBound和OutBound是什么东西

**Comment by draveness** [comment]
> @Devying
> 作者你好，有没有用过weavescope 它的监控图里面的InBound和OutBound是什么东西

没有，Deployment 跟 weavescope 的关系是啥？

**Comment by bzd111** [comment]
两个小问题：
1、maxTotalPods = deployment.Spec.Replicas + maxSurge
少加了maxSurge
2、相关文章没有展示 {% include related/distributed-system.md %}

**Comment by LiuWei-Luke** [comment]
刚在看尚硅谷k8s视频的时候发现，有一节deploy的课里用的图很眼熟，过来一找，果然是你这里的。。。。

**Comment by jxlwqq** [comment]
typo: 

家下来我们将开始介绍该控制器的实现原理。

接下来我们将开始介绍该控制器的实现原理。

**Comment by wgjak47** [comment]
相关文章那里html渲染好像失效了，firefox和chrome都可以复现

**Comment by double12gzh** [comment]
麻烦问一下，咱们这个是基于哪个版本的 k8s 呢，发现文中 `同步` 这一节有提到如何获取 `rs` 时，说会 `查找集群中的全部 ReplicaSet` 似乎不是特别的准确。

```bash
根据传入的键获取 Deployment 资源；
调用 getReplicaSetsForDeployment 获取集群中与 Deployment 相关的全部 ReplicaSet；
查找集群中的全部 ReplicaSet；
```

我基于 1.20 看源码时，获取 `rs` 时，首先会获取与 `deployment` 同 `namespace` 的 `rs`。
https://github.com/kubernetes/kubernetes/blob/0424c7c74d926b4fe3193059e003e9056b429d28/pkg/controller/deployment/deployment_controller.go#L506
